pkg/kp/rcstore/consul_store.go

```diff
@@ -301,34 +312,48 @@ func (s *consulStore) Watch(rc *fields.RC, quit <-chan struct{}) (<-chan struct{
 	errors := make(chan error)
 	input := make(chan *api.KVPair)
 
-	go consulutil.WatchSingle(kp.RCPath(rc.ID.String()), s.kv, input, quit, errors)
-
-	go func() {
-		defer close(updated)
-		defer close(errors)
-
-		for kvp := range input {
-			if kvp == nil {
-				// seems this RC got deleted from under us. quitting
-				// would be unexpected, so we'll just wait for it to
-				// reappear in consul
-				continue
+	rcp, err := rcPath(rc.ID)
+	if err != nil {
+		go func() {
```

Comment:
There's an interesting pattern you see when working with futures that applies here too.  Even though the method is returning a placeholder that would otherwise be filled asynchronously, if the method already has the value, just return a placeholder that already contains the value. As it applies here:

    errors := make(chan error, 1)
    errors <- err
    close(errors)
    return updated, errors